### ISSUE TITLE
Software packaging problems

### ISSUE THREAD

**Issue by abigcoder04** [QuantumPackage/qp2#224]
Dear bro，
After I successfully installed the qp software in the directory /usr/local/qp/qp2 and successfully tested the case, I now want to package the qp and put it in the directory /root I specified, and delete the installed qp package in the directory /usr/local/qp/qp2 at the same time. Now I go to the /root directory to directly test the qp software. He will prompt me to find some files in the/usr/local/qp/qp2 directory. How to solve this problem. I installed ocaml using ./configure -i ocaml, I recompile directly in the/root directory, and the error is as follows：

 System file /usr/local/qp/qp2/external/ocaml-bundle/bootstrap/bin/ocamlc, which package ocaml-bootstrap.4.11.1 depends upon, no longer exists.
        The package has been marked as removed, and opam will try to reinstall it if necessary, but you should reinstall its system dependencies first.
Finished, 0 targets (0 cached) in 00:00:00.
Exception
  Sys_error("/root/pkg/qp-2.1.2-hpc/qp2/ocaml/qptypes_generator.byte: No such file or directory").
Compilation unsuccessful after building 0 targets (0 cached) in 00:00:00.

I would like to ask how to package the qp software to run in other directories or on other machines. I hope to get your reply.

**Comment by scemama** [comment]
See discussions of issue #180. There are links to singularity build files which could be good starting points for you.

**Comment by abigcoder04** [comment]
I took a closer look and found that the developer did not use the automatic build method to install ocaml and reported an error while making. Later, the following comment area gave him a solution that still used ./configure -i ocaml.

**Comment by abigcoder04** [comment]
Hello, if I finished packaging, I re-download install ocaml package path is feasible with ./configure -i ocaml?

**Comment by scemama** [comment]
Hello, it should work.

**Comment by abigcoder04** [comment]
Thank you for your answers, scemama!

**Comment by abigcoder04** [comment]
hello, scemama!
I tried retesting qp on paths other than the installation path, and I have removed the qp package for the installation path, and I tried rerunning. / configure -i ocaml on other paths. The following errors were encountered:

[ERROR] System file /root/install/qp2-2.1.2/external/ocaml-bundle/bootstrap/bin/ocamlc, which package ocaml-bootstrap.4.11.1 depends upon, no longer exists.
        The package has been marked as removed, and opam will try to reinstall it if necessary, but you should reinstall its system dependencies first.

**Comment by abigcoder04** [comment]
hello, scemama!
I tried to package the installed qp and move it to another server and modified it. The path problem in the /bin directory getopt-sample, ocamlbuild, ocamlbuild.byte, ocamlbuild.native. And I also modified the path problem in the /etc directory 00.qp_root.rc, and changed it to the directory where I want to run the currently packaged qp. At this time, I execute ./bin/qpsh can enter the encoding mode of qp. But i can't execute qp create_ezfio -b 6-31g hcn.xyz -o hcn, the system can not recognize. I want to ask if there are other places where the path I have not changed. I hope to get your reply.

**Comment by scemama** [comment]
Hello,
if the `qp` command is not recognized, it is that the `quantum_package.rc` file has not been sourced in the current bash.
Can you tell more about "the system can not recognize"? Can you send the error you get?

**Comment by abigcoder04** [comment]
I currently package the qp package under the installation path to other paths, and delete the successfully installed qp package under the installation path. Then I have modified the path problems in getopt-sample, ocamlbuild, ocamlbuild.byte, ocamlbuild.native and 00.qp _ root.rc under the new directory qp root directory, and then I execute source quantum_ package.rc. I then execute./bin/qpsh to get into qp encoding mode, and then I execute qp create_ezfio-b 6-31g hcn.xyz-o hcn to get into trouble. Normally if I execute the hcn.xyz example, I execute qp create_ezfio-b 6-31g hcn.xyz-o hcn to generate the hcn file, but in this case the error: 
bash : qp_create_ezfio : command not found

**Comment by abigcoder04** [comment]
hello, scemama! 
After I reinstalled ocaml in the new directory, I tried to re-make qp, and the final error of make was as follows :
-------------------------------------------------------------------------------------------------------------------------------------
Sys_error("/root/pkg/quantumpackage-2.1.2-hpc/qp2-2.1.2/ocaml/qptypes_generator.byte: No such file or directory").
-------------------------------------------------------------------------------------------------------------------------------------
File qptypes_generator.byte I have generated in the installation directory, packaged to run under other machines encountered this problem, I would like to ask how this file error should be resolved.  I hope to get your reply.
